crypto/vrf/vrf.go

```diff
@@ -60,15 +59,15 @@ func GenerateKey(rnd io.Reader) (pk *PublicKey, sk *PrivateKey, err error) {
 	pkP.ToBytes(&pkBytes)
 
 	copy(sk[32:], pkBytes[:])
-	copy(pk[:], pkBytes[:])
-	return pk, sk, err
+	return
 }
 
-func (sk *PrivateKey) Public() []byte {
-	return ed25519.PrivateKey(sk[:]).Public().(ed25519.PublicKey)
+func (sk *PrivateKey) Public() (PublicKey, bool) {
+	pk, ok := ed25519.PrivateKey(*sk).Public().(ed25519.PublicKey)
+	return PublicKey(pk), ok
 }
 
-func (sk *PrivateKey) expandSecret() (x, skhr *[32]byte) {
+func (sk PrivateKey) expandSecret() (x, skhr *[32]byte) {
```

Comment:
My preference for pointers here is that I've made the mistake of trying to mutate the value, not remembering that these methods are syntactic sugar, and wondering why my structure wasn't updating.  But that's just me not being very bright.

I'm fine with it either way, under the condition that we change them everywhere to be the same.